src/cloudStorage.ts

```diff
@@ -347,6 +353,9 @@ export class CloudStorage implements CommonStorage {
     toBucket?: string,
   ): Promise<void> {
     const filePaths = await this.getFileNames(bucketName, { prefix })
+    if (filePaths.length === 0) {
```

Comment:
I'd be fine to remove this statement and let it fall through into what you added above. This way we would get the debug message in both cases and we don't do anything extra like external api calls etc :)

Reply:
Yes, sounds good! :D